src/emcee/utils.py

```diff
@@ -60,3 +60,47 @@ def sample_ellipsoid(p0, covmat, size=1):
     return np.random.multivariate_normal(
         np.atleast_1d(p0), np.atleast_2d(covmat), size=size
     )
+
+try:
+    Generator = np.random.Generator
+except AttributeError:
+    Generator = None
+
+def rng_integers(gen, low, **kwargs):
+    """
+    Generate integers from a Generator or RandomState.
+
+    :param gen: Generator or RandomState object
+    :param low: Passed as first argument to gen.integers() or to
+        gen.randint() depending of its type.
+    :param kwargs: Passed to gen.integers() or to gen.randint()
+        depending on its type.
+
+    """
+    if Generator is not None and isinstance(gen, Generator):
+        return gen.integers(low, **kwargs)
+    return gen.randint(low, **kwargs)
+
+def _check_random_state(seed, state):
+    """Check seed argument and set RandomState.
+
+    Based on scikit-learn utils/validation.py.
+    """
+    if isinstance(seed, (int, np.integer)) or seed is None:
+        random = np.random.mtrand.RandomState()
+        random.set_state(state)
+        if seed is not None:
+            random.seed(seed)
+        return random
+    if isinstance(seed, np.random.RandomState):
+        return seed
+    try:
+    # Generator is only available in numpy >= 1.17
+        if isinstance(seed, np.random.Generator):
+            return seed
```

Comment:
Could/should the state of the Generator also be set? e.g.
```
if isinstance(seed, Generator):
    if isinstance(state, dict):
        seed.bit_generator.state = state
    return seed
```
? Checking that state is a dict isn't going to solve all issues if state isn't what the bit generator is expecting, but it'll catch most things.

Reply:
We could, that depends on what do we want the seed argument to do. Currently the state is only used if the seed in None where the exact same state is used, or if it is an int, in which case the state is used and reseeded. Using RandomState of Generator overwrites the stored state to use directly the passed object. 

Also, the behaviour does not depend on reusing a backend or not, maybe it should?